### PR TITLE
kvm: fix backup volume snapshot fails on RBD storage

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes the issue that volume snapshot fails on RBD storage with the following error

qemu-img: Could not open 'driver=raw,file.filename=rbd:cloudstack/test_wei.img:mon_host=10.0.32.254:auth_supported=cephx:id=cloudstack:key=AQDwHTNjjHXRKRAAJb+AToFr6x4a1AvKUc4Ksg==:rbd_default_format=2:client_mount_timeout=30': Could not open 'rbd:cloudstack/test_wei.img:mon_host=10.0.32.254:auth_supported=cephx:id=cloudstack:key=AQDwHTNjjHXRKRAAJb+AToFr6x4a1AvKUc4Ksg==:rbd_default_format=2:client_mount_timeout=30': No such file or directory

However, it works without using image options

Therefore, do not pass the image options if the image format is not QCOW2 and LUKS.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@mlsorensen 
This PR has some changes in the file QemuImageOptions.java which was introduced by your nice PR (Volume encryption feature #6522).
can you review this PR ?

**Comment by weizhouapache** [comment]
@blueorangutan package

File: plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImageOptions.java [hunk 1]
```diff
@@ -56,13 +62,15 @@ public QemuImageOptions(QemuImg.PhysicalDiskFormat format, String filePath, Stri
             }
         }
         if (format != null) {
-            params.put("driver", format.toString());
+            params.put(DRIVER, format.toString());
+            this.format = format;
         }
```
Comment (DaanHoogland) [inline plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImageOptions.java:67]: we can change this to 
```suggestion
        setFormat(format);
```

**Review by DaanHoogland — APPROVED** [review]
clgtm, test updated for this, though I am not clear on whether this can be tested more (unit, integration or monkey)

File: plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImageOptions.java (hunk 1 above)
Comment (weizhouapache) [reply plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImageOptions.java]: @DaanHoogland 
good point. thanks
applied

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

File: plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImageOptions.java [hunk 2]
```diff
@@ -27,6 +29,10 @@ public class QemuImageOptions {
     private static final String FILENAME_PARAM_KEY = "file.filename";
     private static final String LUKS_KEY_SECRET_PARAM_KEY = "key-secret";
     private static final String QCOW2_KEY_SECRET_PARAM_KEY = "encrypt.key-secret";
+    private static final String DRIVER = "driver";
+
+    private QemuImg.PhysicalDiskFormat format;
+    private static final List<QemuImg.PhysicalDiskFormat> supportedFormats = Arrays.asList(QemuImg.PhysicalDiskFormat.QCOW2, QemuImg.PhysicalDiskFormat.LUKS);
```
[the remark this answers is not in the document]
Comment (weizhouapache) [reply plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImageOptions.java]: thanks @GutoVeronezi 
done

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**